### PR TITLE
Add check-network sub-command

### PR DESCRIPTION
**Pull request by alyssa1303**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aks-preview/azext_aks_preview/_client_factory.py [hunk 1]
```diff
@@ -65,8 +65,8 @@ def cf_trustedaccess_role_binding(cli_ctx, *_):
 
 def get_compute_client(cli_ctx, *_):
     return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_COMPUTE)
-
-
+
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_client_factory.py:68]: please remove these extra spaces

File: src/aks-preview/azext_aks_preview/_params.py [hunk 2]
```diff
@@ -899,7 +899,13 @@ def load_arguments(self, _):
     with self.argument_context('aks mesh disable-ingress-gateway') as c:
         c.argument('ingress_gateway_type',
                    arg_type=get_enum_type(ingress_gateway_types))
-
+        
+    with self.argument_context('aks check-network vmss') as c:
+        c.argument('cluster_name', options_list=['--name', '-n'], required=True, help='Name of the managed cluster.')
+        c.argument('resource_group', options_list=['--resource-group', '-g'], required=True, help='Name of the resource group.')
+        c.argument('location', options_list=['--location', '-l'], required=True, help='Location of the managed cluster.')
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_params.py:906]: could get location from mc

File: src/aks-preview/azext_aks_preview/commands.py [hunk 3]
```diff
@@ -14,6 +14,7 @@
     cf_trustedaccess_role,
     cf_trustedaccess_role_binding,
 )
+from azure.cli.command_modules.vm._client_factory import cf_vmss_run_commands
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/commands.py:17]: no need to import this

File: src/aks-preview/azext_aks_preview/commands.py [hunk 4]
```diff
@@ -118,6 +119,12 @@ def load_command_table(self, _):
         client_factory=cf_trustedaccess_role_binding
     )
 
+    vmss_run_sdk = CliCommandType(
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/commands.py:122]: no need to declare this

File: src/aks-preview/azext_aks_preview/commands.py [hunk 5]
```diff
@@ -267,3 +274,7 @@ def load_command_table(self, _):
             'aks_mesh_disable_ingress_gateway',
             supports_no_wait=True,
             confirmation=True)
+        
+    with self.command_group('aks check-network', vmss_run_sdk, client_factory=cf_vmss_run_commands) as g:
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/commands.py:278]:
```suggestion
    with self.command_group('aks check-network', managed_clusters_sdk, client_factory=cf_managed_clusters) as g:
```

File: src/aks-preview/azext_aks_preview/custom.py [hunk 6]
```diff
@@ -2477,3 +2481,70 @@ def _aks_mesh_update(
...
+    if not node_name:
+        if not _which("kubectl"):
+            raise ValidationError("Can not find kubectl executable in PATH")
+        
+        fd, browse_path = tempfile.mkstemp()
+        try:
+            aks_get_credentials(cmd, client, resource_group, cluster_name, admin=False, path=browse_path)
+
+            cmd = f"kubectl --kubeconfig {browse_path} get nodes"
+            output = subprocess.Popen(cmd, shell=True, stderr=subprocess.PIPE)
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/custom.py:2523]: `subprocess.checkout` might be better

File: src/aks-preview/azext_aks_preview/custom.py [hunk 7]
```diff
@@ -2477,3 +2481,70 @@ def _aks_mesh_update(
...
+            raise ValidationError("Can not get node name from cluster: {}".format(ex))
+        if output:
+            result, _ = output.communicate()
+            cluster_info = json.loads(result)
+            node_name = str(cluster_info["items"][0]["metadata"]["name"])
+        else:
+            raise ValidationError("Failed to get node name from cluster")
+    index = node_name.find("vmss")
+    vmss_name = node_name[0:index+4]
+    instance_id = int(node_name[index+4:])
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/custom.py:2534]:
```suggestion
    instance_id = node_name[index+4:]
```

in vmss api, it takes this as a string

File: src/aks-preview/azext_aks_preview/custom.py [hunk 8]
```diff
@@ -2477,3 +2481,70 @@ def _aks_mesh_update(
...
+    node_resource_group = "MC_{0}_{1}_{2}".format(resource_group, cluster_name, location)
+    print("Check parameters vmss_name: {0}, instance_id: {1}, node_resource_group: {2}".format(vmss_name, instance_id, node_resource_group))
+
+    from azure.cli.command_modules.vm.custom import vmss_run_command_invoke
+    try:
+        command_result_poller = vmss_run_command_invoke(cmd, node_resource_group, vmss_name, command_id="RunShellScript", scripts="bash /usr/local/bin/check-outbound-network.sh", instance_id=instance_id)
+        command_result = command_result_poller.result()
+        display_status = command_result.value[0].displayStatus
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/custom.py:2542]:
```suggestion
        display_status = command_result.value[0].display_status
```
see https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/compute/azure-mgmt-compute/azure/mgmt/compute/v2023_03_01/models/_models_py3.py#L3743

File: src/aks-preview/azext_aks_preview/custom.py [hunk 9]
```diff
@@ -2477,3 +2481,70 @@ def _aks_mesh_update(
...
+            raise ValidationError("Failed to get node name from cluster")
+    index = node_name.find("vmss")
+    vmss_name = node_name[0:index+4]
+    instance_id = int(node_name[index+4:])
+    node_resource_group = "MC_{0}_{1}_{2}".format(resource_group, cluster_name, location)
+    print("Check parameters vmss_name: {0}, instance_id: {1}, node_resource_group: {2}".format(vmss_name, instance_id, node_resource_group))
+
+    from azure.cli.command_modules.vm.custom import vmss_run_command_invoke
+    try:
+        command_result_poller = vmss_run_command_invoke(cmd, node_resource_group, vmss_name, command_id="RunShellScript", scripts="bash /usr/local/bin/check-outbound-network.sh", instance_id=instance_id)
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/custom.py:2540]: It's hard to re-use this func as it doesn't load the models with necessary kwargs (missing resource_type and operation_group) for outer func to use.

Could call the operation by yourself. May refer to commit [#88db6ef](https://github.com/FumingZhang/azure-cli-extensions/commit/88db6ef7bea0416fdc61fd19ca409780ed9d47b8).

File: src/aks-preview/azext_aks_preview/custom.py [hunk 10]
```diff
@@ -2477,3 +2481,70 @@ def _aks_mesh_update(
...
+                      location,
+                      node_name, 
+                      no_wait=False, 
+                      aks_custom_headers=None):
+    # Get a random node if node_name is not specified
+    if not node_name:
+        if not _which("kubectl"):
+            raise ValidationError("Can not find kubectl executable in PATH")
+        
+        fd, browse_path = tempfile.mkstemp()
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/custom.py:2518]: better close the fd and remove the temp file finally (via try/finally block)

**Review by FumingZhang — COMMENTED** [review]
LGTM, please fix lint issues and update history.rst and setup.py if you want to release a new version of aks-preview.

**Comment by alyssa1303** [comment]
Closing due to error in rebase so have to open a new one

**Pull request closed without merging**